pkg/sql/sql_cursor.go

```diff
@@ -13,24 +13,169 @@ package sql
 import (
 	"context"
 
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgcode"
+	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgerror"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
 	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
+	"github.com/cockroachdb/errors"
 )
 
 // DeclareCursor implements the DECLARE statement.
 // See https://www.postgresql.org/docs/current/sql-declare.html for details.
 func (p *planner) DeclareCursor(ctx context.Context, s *tree.DeclareCursor) (planNode, error) {
-	return nil, unimplemented.NewWithIssue(41412, "DECLARE CURSOR")
+	if p.autoCommit {
+		return nil, pgerror.Newf(pgcode.NoActiveSQLTransaction, "DECLARE CURSOR can only be used in transaction blocks")
+	}
+
+	if s.Hold {
+		return nil, unimplemented.NewWithIssue(41412, "DECLARE CURSOR WITH HOLD")
+	}
+	if s.Binary {
+		return nil, unimplemented.NewWithIssue(41412, "DECLARE BINARY CURSOR")
+	}
+	if s.Scroll == tree.Scroll {
+		return nil, unimplemented.NewWithIssue(41412, "DECLARE SCROLL CURSOR")
+	}
+	if s.Sensitivity != tree.UnspecifiedSensitivity {
```

Comment:
i think we can just let this "pass through":

```
ASENSITIVE
INSENSITIVE
Cursor sensitivity determines whether changes to the data underlying the cursor, done in the same transaction, after the cursor has been declared, are visible in the cursor. INSENSITIVE means they are not visible, ASENSITIVE means the behavior is implementation-dependent. A third behavior, SENSITIVE, meaning that such changes are visible in the cursor, is not available in PostgreSQL. In PostgreSQL, all cursors are insensitive; so these key words have no effect and are only accepted for compatibility with the SQL standard.

Specifying INSENSITIVE together with FOR UPDATE or FOR SHARE is an error.
```

https://www.postgresql.org/docs/current/sql-declare.html

Reply:
Done.